src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,562 @@
+            var2 = ((long)_calibrationData.DigP8 * p) >> 19;
+            p = ((p + var1 + var2) >> 8) + ((long)_calibrationData.DigP7 << 4);
+            return p;
+        }
+
+        /// <summary>
+        ///  Reads an 8 bit value from a register
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        public byte Read8(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                _i2cDevice.WriteByte(register);
+                var value = _i2cDevice.ReadByte();
+                return value;
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        ///  Reads a 16 bit value over I2C 
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        public ushort Read16(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                byte[] readBuffer = { 0x00, 0x00 };
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(readBuffer);
+                int h = readBuffer[1] << 8;
+                int l = readBuffer[0];
+                ushort value = (ushort)(h + l);
+
+                return value;
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        ///  Reads a 24 bit value over I2C 
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        public uint Read24(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                byte[] readBuffer = { 0x00, 0x00, 0x00 };
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(readBuffer);
+                uint value = readBuffer[2];
+                value <<= 8;
+                value = readBuffer[1];
+                value <<= 8;
+                value = readBuffer[0];
+
+                return value;
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public void Dispose()
+        {
+            if (_i2cDevice != null)
+            {
+                _i2cDevice.Dispose();
+                _i2cDevice = null;
+            }
+        }
+
+        /// <summary>
+        /// BMP280s power modes
+        /// </summary>
+        public enum PowerMode : byte
```

Comment:
I would add one extra name in your namespace, (Iot.Device.Bmp280 instead of just Iot.Device) and then move these three enums to their own individual files in order to get these class a bit more clean. That is also the pattern that we have been following with these device bindings.